brave/src/main/java/brave/internal/PredefinedPropagationFields.java

```diff
@@ -78,12 +78,12 @@ public String get(int index) {
   }
 
   public final void put(int index, String value) {
-    if (index >= fieldNames.length) return;
+    if (index >= names.length) return;
 
     synchronized (this) {
```

Comment:
drifted to ExtraFields but will comment now!

putAllIfAbsent is protected and used before the input is returned anywhere. I'll add a javadoc comment. Regardless I made a path that doesn't synchronize even if it would be uncontended. Good idea.